### PR TITLE
fix(ci): validate tests exit code after reading the container logs

### LINKED ISSUE #3664: sync_past_mandatory_checkpoint_mainnet fails, but CI job succeeds
## Motivation

The `sync_past_mandatory_checkpoint_mainnet` test is currently failing:
>    0: stdout of command did not contain any matches for the given regex
>
> Match Regex:
>   "stopping at configured height"

https://github.com/ZcashFoundation/zebra/runs/5329552803?check_suite_focus=true#step:10:199

But the CI job succeeds anyway.

### Tasks

- [x] fix the test harness so the test passes
    - done in #3688
- [x] make the CI job fail when the test fails
- [x] check other CI jobs for the same bug

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

When deploying Tests to GCP we're doing it with docker containers, which requires to read the logs from GitHub Actions, but after this process ends we don't know the actual exit code generated inside the test, so we're not able to register any panics or errors within it.

Fixes #3664 #3500  

## Solution
- Create a step that reads the resulting outcome from the tests or the container, and use it as a way to fail or pass the check.

## Review
Not ready for review

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -461,6 +461,18 @@ jobs:
...
 
+      - name: Check exit code
+        id: check-exit-code
+        run: |
+          gcloud compute ssh \
+          sync-checkpoint-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} \
+          --zone ${{ env.ZONE }} \
+          --quiet \
+          --ssh-flag="-o ServerAliveInterval=5" \
+          --command="exit_code=$(docker wait ${{ env.ZEBRA_CONTAINER }}) && if [ $exit_code -eq "0" ]; then exit 0; else exit 1; fi"
```
Comment (teor2345) [inline .github/workflows/test.yml]: Question: is there any reason to turn all non-zero edit codes into 1?

```suggestion
          --command="docker wait ${{ env.ZEBRA_CONTAINER }}"
```

Comment (gustavovalverde) [reply .github/workflows/test.yml]: That was my first try, but I stumbled with this issue https://github.com/docker/cli/pull/2244#issue-546334620. The exit code was not being sent to GitHub Actions

Comment (teor2345) [reply .github/workflows/test.yml]: Wow, that's a terrible interface, sorry you have to deal with that!

Comment (teor2345) [reply .github/workflows/test.yml]: Maybe we could put a comment in the file, so we don't accidentally simplify the code and break it?

**Comment by gustavovalverde** [comment]
@teor2345 @dconnolly this is finally working as expected, I made a few tests forcing a panic and others without it

File: .github/workflows/test-full-sync.yml [hunk 2]
```diff
@@ -221,6 +221,20 @@ jobs:
...
+          --zone ${{ env.ZONE }} \
+          --quiet \
+          --ssh-flag="-o ServerAliveInterval=5" \
+          --command="docker wait ${{ env.CONTAINER_NAME }}")
+
+          if [[ ${EXIT_CODE} -eq "0" ]]; then
+            exit 0
+          else
+            exit 1
+          fi
```
Comment (dconnolly) [inline .github/workflows/test-full-sync.yml]: Can we do just `exit ${EXIT_CODE}`?

Comment (gustavovalverde) [reply .github/workflows/test-full-sync.yml]: Fixed with [imp(ci): use single line exit](https://github.com/ZcashFoundation/zebra/pull/3968/commits/d6c37c167384d92297dbe87dcc1ddeef8303fdf0)

**Review by dconnolly — DISMISSED** [review]
Baller 🎉

**Review by teor2345 — APPROVED** [review]
Looks good.

I just added a CODEOWNERS cleanup in commit 0fc54f6950bddce1e02dd923e6306a8ae7fe3031.

**Pull request merged**